frontend/src/metabase/admin/settings/components/widgets/AdminSettingInput.tsx

```diff
@@ -186,6 +190,14 @@ export function BasicAdminSettingInput({
   }
 }
 
+// some radio inputs actually should produce boolean values
+function toBoolean(value: string | boolean): boolean | string {
```

Comment:
I'm not sure how this function knows if it should convert the value to a boolean or not?

I believe `newValue` is always a string, so toBoolean always gets a string 
` onChange={(newValue) => handleChange(toBoolean(newValue))}`

Reply:
If the string is exactly "true" or "false", it makes it a boolean, otherwise, it leaves it as-is

Reply:
Ah yes I get it, for some reason this line was not clicking for me `value === "true"`

How do you feel about the assumption that true/false strings should always be converted to booleans?

Reply:
For all we know there's some other whacky setting where we've set a string value to "true" and/or "false" :sweat_smile: